Cyclistic Case Study

1. Scenario

Cyclistic is a bike-share company in Chicago. It offers flexibility of its pricing plans: single-ride passes, full-day passes, and annual memberships. Customers who purchase single-ride or full-day passes are referred to as casual riders. Customers who purchase annual memberships are Cyclistic members. 
Cyclistic’s finance analysts have concluded that annual members are much more profitable than casual riders. The director of marketing believes the company’s future success depends on maximizing the number of annual memberships. 


2. The purpose of the case study

To identify and understand the differences in the usage patterns of Cyclistic bikes between annual members and casual riders. By doing so, we aim to develop an effective marketing strategy that encourages casual riders to upgrade to the annual subscription. Our insights from this study can drive business decisions by informing targeted marketing campaigns and potentially increasing revenue through increased subscription rates.


3. Ask phase

Our main objective (Business task) is to develop marketing strategies that effectively convert casual riders into annual members. The success of this project will be crucial for the growth of Cyclistic bike-share program, and it requires the collaboration of the stakeholders, which include Lily Moreno, the director of marketing and our manager, who will be responsible for implementing our recommendations; alongside the Cyclistic marketing analytics team, who will provide us with valuable data to analyze and report, and the Cyclistic executive team, who make the final decisions, will need to approve our proposed marketing program. By working together, we can create a successful marketing strategy that will drive growth for Cyclistic.


4. Prepare Phase

Motivate International Inc. has made the historical trip data of Cyclistic available under this license (https://www.divvybikes.com/data-license-agreement). This data is publicly available and can be used to explore how different customer types are using Cyclistic bikes. The open-source public data consists of 10 datasets that display data from April 2020 to December 2020, along with one file containing data from the first quarter. The files can be downloaded as compressed .csv files from [Cyclistic_Dataset](https://www.kaggle.com/datasets/timgid/cyclistic-dataset-google-certificate-capstone). 



5. Process phase

This step will prepare the data for analysis. All the csv files will be merged into one file to improve workflow

In [1]:
#installing tidyverse package.
#tidyverse is a collection of R packages specifically designed for working with data. 
install.packages("tidyverse")


Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



In [2]:
#installing tidyverse library.
#It’s a standard library for most data analysis.
library(tidyverse)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.2     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.2     ✔ tibble    3.2.1
✔ lubridate 1.9.2     ✔ tidyr     1.3.0
✔ purrr     1.0.1     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


In [3]:
#concatanating files into one dataframe

# List the file names of the CSV files to be concatenated
csv_files <- list.files(path = "../input", recursive = TRUE, full.names=TRUE)

# Initialize an empty data frame to store the concatenated data
concatenated_data <- data.frame()

# Iterate over each CSV file
for (file in csv_files) {
  # Read the CSV file and append its data to the concatenated_data data frame
  data <- read.csv(file)
  concatenated_data <- rbind(concatenated_data, data)
}

# Write the concatenated data to a new CSV file
cyclistic_merged <- do.call(rbind, lapply(csv_files, read.csv))

#using head() function to display the first few rows of a data frame
head(cyclistic_merged)

ride_id          rideable_type started_at          ended_at           
1 A847FADBBC638E45 docked_bike   2020-04-26 17:45:14 2020-04-26 18:12:03
2 5405B80E996FF60D docked_bike   2020-04-17 17:08:54 2020-04-17 17:17:03
3 5DD24A79A4E006F4 docked_bike   2020-04-01 17:54:13 2020-04-01 18:08:36
4 2A59BBDF5CDBA725 docked_bike   2020-04-07 12:50:19 2020-04-07 13:02:31
5 27AD306C119C6158 docked_bike   2020-04-18 10:22:59 2020-04-18 11:15:54
6 356216E875132F61 docked_bike   2020-04-30 17:55:47 2020-04-30 18:01:11
  start_station_name                  start_station_id
1 Eckhart Park                        86              
2 Drake Ave & Fullerton Ave           503             
3 McClurg Ct & Erie St                142             
4 California Ave & Division St        216             
5 Rush St & Hubbard St                125             
6 Mies van der Rohe Way & Chicago Ave 173             
  end_station_name            end_station_id start_lat start_lng end_lat
1 Lincoln Ave & Diversey Pkwy 152            41.8964   -87.6610  41.9322
2 Kosciuszko Park             499            41.9244   -87.7154  41.9306
3 Indiana Ave & Roosevelt Rd  255            41.8945   -87.6179  41.8679
4 Wood St & Augusta Blvd      657            41.9030   -87.6975  41.8992
5 Sheridan Rd & Lawrence Ave  323            41.8902   -87.6262  41.9695
6 Streeter Dr & Grand Ave     35             41.8969   -87.6217  41.8923
  end_lng  member_casual
1 -87.6586 member       
2 -87.7238 member       
3 -87.6230 member       
4 -87.6722 member       
5 -87.6547 casual       
6 -87.6120 member

In [4]:
#Data cleaning

#Removing duplicates

# removing duplicates based on the column ride_id in the cyclistic_merged data frame
cyclistic_no_dups <- cyclistic_merged[!duplicated(cyclistic_merged$ride_id), ]
print(paste("Removed", nrow(cyclistic_merged) - nrow(cyclistic_no_dups), "duplicated rows"))



[1] "Removed 209 duplicated rows"


In [5]:
install.packages("tidyr")

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



In [6]:
#removing missing values (NA)

cyclistic_merged <- drop_na(cyclistic_merged)
head(cyclistic_merged)


ride_id          rideable_type started_at          ended_at           
1 A847FADBBC638E45 docked_bike   2020-04-26 17:45:14 2020-04-26 18:12:03
2 5405B80E996FF60D docked_bike   2020-04-17 17:08:54 2020-04-17 17:17:03
3 5DD24A79A4E006F4 docked_bike   2020-04-01 17:54:13 2020-04-01 18:08:36
4 2A59BBDF5CDBA725 docked_bike   2020-04-07 12:50:19 2020-04-07 13:02:31
5 27AD306C119C6158 docked_bike   2020-04-18 10:22:59 2020-04-18 11:15:54
6 356216E875132F61 docked_bike   2020-04-30 17:55:47 2020-04-30 18:01:11
  start_station_name                  start_station_id
1 Eckhart Park                        86              
2 Drake Ave & Fullerton Ave           503             
3 McClurg Ct & Erie St                142             
4 California Ave & Division St        216             
5 Rush St & Hubbard St                125             
6 Mies van der Rohe Way & Chicago Ave 173             
  end_station_name            end_station_id start_lat start_lng end_lat
1 Lincoln Ave & Diversey Pkwy 152            41.8964   -87.6610  41.9322
2 Kosciuszko Park             499            41.9244   -87.7154  41.9306
3 Indiana Ave & Roosevelt Rd  255            41.8945   -87.6179  41.8679
4 Wood St & Augusta Blvd      657            41.9030   -87.6975  41.8992
5 Sheridan Rd & Lawrence Ave  323            41.8902   -87.6262  41.9695
6 Streeter Dr & Grand Ave     35             41.8969   -87.6217  41.8923
  end_lng  member_casual
1 -87.6586 member       
2 -87.7238 member       
3 -87.6230 member       
4 -87.6722 member       
5 -87.6547 casual       
6 -87.6120 member

In [7]:
#calculating the ride time in minutes (ride_time_m) by subtracting the started_at from the ended_at column
#converting the result to minutes using the as.numeric() function. 
#summary() function is used to generate a summary of the ride_time_m column.
cyclistic_no_dups <- cyclistic_no_dups %>%
  mutate(ended_at = as.POSIXct(ended_at),
         started_at = as.POSIXct(started_at),
         ride_time_m = as.numeric(ended_at - started_at) / 60)

summary(cyclistic_no_dups$ride_time_m)


     Min.   1st Qu.    Median      Mean   3rd Qu.      Max. 
-29049.97      7.63     14.07     26.52     25.93 156450.40 

In [8]:
#Converting data frame to a datetime format
#This will be represented by POSIXct objects. 
#This allows for easier manipulation and analysis of the date and time values in subsequent data operations.

#Converting the "started_at" and "ended_at" columns in the data frame cyclistic_no_dups to POSIXct (POSIXct is a class representing date and time values) format. 
#It uses the as.POSIXct() function to perform the conversion

cyclistic_no_dups$started_at <- as.POSIXct(cyclistic_no_dups$started_at, "%Y-%m-%d %H:%M:%S")
cyclistic_no_dups$ended_at <- as.POSIXct(cyclistic_no_dups$ended_at, "%Y-%m-%d %H:%M:%S")

#creating a column called “ride_length, calculating the length of the rides.
cyclistic_no_dups$ride_length <- as.POSIXct(cyclistic_no_dups$ended_at) - as.POSIXct(cyclistic_no_dups$started_at)

head(cyclistic_no_dups)


ride_id          rideable_type started_at          ended_at           
1 A847FADBBC638E45 docked_bike   2020-04-26 17:45:14 2020-04-26 18:12:03
2 5405B80E996FF60D docked_bike   2020-04-17 17:08:54 2020-04-17 17:17:03
3 5DD24A79A4E006F4 docked_bike   2020-04-01 17:54:13 2020-04-01 18:08:36
4 2A59BBDF5CDBA725 docked_bike   2020-04-07 12:50:19 2020-04-07 13:02:31
5 27AD306C119C6158 docked_bike   2020-04-18 10:22:59 2020-04-18 11:15:54
6 356216E875132F61 docked_bike   2020-04-30 17:55:47 2020-04-30 18:01:11
  start_station_name                  start_station_id
1 Eckhart Park                        86              
2 Drake Ave & Fullerton Ave           503             
3 McClurg Ct & Erie St                142             
4 California Ave & Division St        216             
5 Rush St & Hubbard St                125             
6 Mies van der Rohe Way & Chicago Ave 173             
  end_station_name            end_station_id start_lat start_lng end_lat
1 Lincoln Ave & Diversey Pkwy 152            41.8964   -87.6610  41.9322
2 Kosciuszko Park             499            41.9244   -87.7154  41.9306
3 Indiana Ave & Roosevelt Rd  255            41.8945   -87.6179  41.8679
4 Wood St & Augusta Blvd      657            41.9030   -87.6975  41.8992
5 Sheridan Rd & Lawrence Ave  323            41.8902   -87.6262  41.9695
6 Streeter Dr & Grand Ave     35             41.8969   -87.6217  41.8923
  end_lng  member_casual ride_time_m ride_length
1 -87.6586 member        26.81667    1609 secs  
2 -87.7238 member         8.15000     489 secs  
3 -87.6230 member        14.38333     863 secs  
4 -87.6722 member        12.20000     732 secs  
5 -87.6547 casual        52.91667    3175 secs  
6 -87.6120 member         5.40000     324 secs

In [9]:
#Creating a column called “day_of_week,” and calculate the day of the week that each ride started using the “WEEKDAY” command 
cyclistic_no_dups$day_of_week <- weekdays(as.Date(cyclistic_no_dups$started_at))
head(cyclistic_no_dups)


ride_id          rideable_type started_at          ended_at           
1 A847FADBBC638E45 docked_bike   2020-04-26 17:45:14 2020-04-26 18:12:03
2 5405B80E996FF60D docked_bike   2020-04-17 17:08:54 2020-04-17 17:17:03
3 5DD24A79A4E006F4 docked_bike   2020-04-01 17:54:13 2020-04-01 18:08:36
4 2A59BBDF5CDBA725 docked_bike   2020-04-07 12:50:19 2020-04-07 13:02:31
5 27AD306C119C6158 docked_bike   2020-04-18 10:22:59 2020-04-18 11:15:54
6 356216E875132F61 docked_bike   2020-04-30 17:55:47 2020-04-30 18:01:11
  start_station_name                  start_station_id
1 Eckhart Park                        86              
2 Drake Ave & Fullerton Ave           503             
3 McClurg Ct & Erie St                142             
4 California Ave & Division St        216             
5 Rush St & Hubbard St                125             
6 Mies van der Rohe Way & Chicago Ave 173             
  end_station_name            end_station_id start_lat start_lng end_lat
1 Lincoln Ave & Diversey Pkwy 152            41.8964   -87.6610  41.9322
2 Kosciuszko Park             499            41.9244   -87.7154  41.9306
3 Indiana Ave & Roosevelt Rd  255            41.8945   -87.6179  41.8679
4 Wood St & Augusta Blvd      657            41.9030   -87.6975  41.8992
5 Sheridan Rd & Lawrence Ave  323            41.8902   -87.6262  41.9695
6 Streeter Dr & Grand Ave     35             41.8969   -87.6217  41.8923
  end_lng  member_casual ride_time_m ride_length day_of_week
1 -87.6586 member        26.81667    1609 secs   Sunday     
2 -87.7238 member         8.15000     489 secs   Friday     
3 -87.6230 member        14.38333     863 secs   Wednesday  
4 -87.6722 member        12.20000     732 secs   Tuesday    
5 -87.6547 casual        52.91667    3175 secs   Saturday   
6 -87.6120 member         5.40000     324 secs   Thursday

In [10]:
#Formatting as a number with no decimals
# 1 = Sunday and 7 = Saturday.
cyclistic_no_dups$day_of_week <- as.integer(factor(cyclistic_no_dups$day_of_week, levels = c("Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday")))
head(cyclistic_no_dups)


ride_id          rideable_type started_at          ended_at           
1 A847FADBBC638E45 docked_bike   2020-04-26 17:45:14 2020-04-26 18:12:03
2 5405B80E996FF60D docked_bike   2020-04-17 17:08:54 2020-04-17 17:17:03
3 5DD24A79A4E006F4 docked_bike   2020-04-01 17:54:13 2020-04-01 18:08:36
4 2A59BBDF5CDBA725 docked_bike   2020-04-07 12:50:19 2020-04-07 13:02:31
5 27AD306C119C6158 docked_bike   2020-04-18 10:22:59 2020-04-18 11:15:54
6 356216E875132F61 docked_bike   2020-04-30 17:55:47 2020-04-30 18:01:11
  start_station_name                  start_station_id
1 Eckhart Park                        86              
2 Drake Ave & Fullerton Ave           503             
3 McClurg Ct & Erie St                142             
4 California Ave & Division St        216             
5 Rush St & Hubbard St                125             
6 Mies van der Rohe Way & Chicago Ave 173             
  end_station_name            end_station_id start_lat start_lng end_lat
1 Lincoln Ave & Diversey Pkwy 152            41.8964   -87.6610  41.9322
2 Kosciuszko Park             499            41.9244   -87.7154  41.9306
3 Indiana Ave & Roosevelt Rd  255            41.8945   -87.6179  41.8679
4 Wood St & Augusta Blvd      657            41.9030   -87.6975  41.8992
5 Sheridan Rd & Lawrence Ave  323            41.8902   -87.6262  41.9695
6 Streeter Dr & Grand Ave     35             41.8969   -87.6217  41.8923
  end_lng  member_casual ride_time_m ride_length day_of_week
1 -87.6586 member        26.81667    1609 secs   1          
2 -87.7238 member         8.15000     489 secs   6          
3 -87.6230 member        14.38333     863 secs   4          
4 -87.6722 member        12.20000     732 secs   3          
5 -87.6547 casual        52.91667    3175 secs   7          
6 -87.6120 member         5.40000     324 secs   5

6. Analyze phase

Descriptive analysis

In [11]:
#Summary Statistics:

# Summary statistics for numerical variables
summary(cyclistic_no_dups$ride_length)

# Summary statistics for categorical variables
table(cyclistic_no_dups$day_of_week)

  Length    Class     Mode 
 3541474 difftime  numeric 


     1      2      3      4      5      6      7 
527220 433696 452858 482008 491278 528235 626179 

In [12]:
#Frequency Distribution:

# Frequency distribution for categorical variables
table(cyclistic_no_dups$rideable_type)
table(cyclistic_no_dups$member_casual)


 classic_bike   docked_bike electric_bike 
        70616       2966113        504745 


 casual  member 
1366550 2174924 

In [13]:
#Measures of Central Tendency and Dispersion:

# Mean and max of ride length
mean_value <- mean(cyclistic_no_dups$ride_length)
max_value <- max(cyclistic_no_dups$ride_length)

# Standard deviation and variance of ride length
sd_value <- sd(cyclistic_no_dups$ride_length)
var_value <- var(cyclistic_no_dups$ride_length)

# Display the outputs with descriptive strings
paste("Mean of ride length:", mean_value)
paste("Max of ride length:", max_value)
paste("Standard deviation of ride length:", sd_value)
paste("Variance of ride length:", var_value)


[1] "Mean of ride length: 1591.39626409794"

[1] "Max of ride length: 9387024"

[1] "Standard deviation of ride length: 22893.4228613153"

[1] "Variance of ride length: 524108810.306995"

In [14]:
#Cross-tabulations:

# Cross-tabulation between rideable_type and member_casual
table(cyclistic_no_dups$rideable_type, cyclistic_no_dups$member_casual)


               
                 casual  member
  classic_bike    11319   59297
  docked_bike   1146005 1820108
  electric_bike  209226  295519

In [15]:
# Install and load reshape2 package
install.packages("reshape2")
library(reshape2)

# Pivot table: Average ride_length for members and casual riders
pivot_table1 <- dcast(cyclistic_no_dups, formula = . ~ member_casual, 
                     value.var = "ride_length", fun.aggregate = mean)

# Pivot table: Average ride_length for users by day_of_week
pivot_table2 <- dcast(cyclistic_no_dups, formula = day_of_week ~ member_casual, 
                     value.var = "ride_length", fun.aggregate = mean)

# Pivot table: Number of rides for users by day_of_week
cyclistic_no_dups$ride_count <- 1
pivot_table3 <- dcast(cyclistic_no_dups, formula = day_of_week ~ member_casual, 
                     value.var = "ride_count", fun.aggregate = sum)

# Print pivot tables
print(pivot_table1)
print(pivot_table2)
print(pivot_table3)


Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)


Attaching package: ‘reshape2’


The following object is masked from ‘package:tidyr’:

    smiths




  .   casual  member
1 . 2770.626 850.462
  day_of_week   casual    member
1           1 3247.111 1085.1372
2           2 2780.279  893.8710
3           3 1882.603  242.5618
4           4 2550.459  884.5852
5           5 2775.929  884.4784
6           6 2713.209  928.1545
7           7 2907.824 1066.9779
  day_of_week casual member
1           1 256148 271072
2           2 141652 292044
3           3 137547 315311
4           4 152524 329484
5           5 162689 328589
6           6 202187 326048
7           7 313803 312376


In [16]:
# Export pivot tables to CSV files
write.csv(pivot_table1, file = "pivot_table1.csv", row.names = FALSE)
write.csv(pivot_table2, file = "pivot_table2.csv", row.names = FALSE)
write.csv(pivot_table3, file = "pivot_table3.csv", row.names = FALSE)


In [17]:
write.csv(cyclistic_no_dups, file = "cyclistic_no_dups.csv", row.names = FALSE)

7. Share Phase

After analyzing the data, it was possible to finally answer the question of "how annual members and casual riders use Cyclistic bikes differently?"

Data visualization using Tableau software, can be accessed at https://public.tableau.com/views/Cyclistic_Bike_Share_Case_Study/Cyclistic_Bike_Share_Case_Study?:language=en-US&:display_count=n&:origin=viz_share_link

Findings:

* The data shows that Members perform more rides than Casual users, around 38% more. However, Casual users use to ride longer distances than Members, around 50% more. The same for the time spent per ride, Casual users use to spend 50% more time riding than Members.
* Also, both categories of users prefer docker bikes.
* The graphics show that for Casual users, on Fridays, Saturday and Sundays there is an increase in the number os rides, which decreases abruptaly on Tuesday, but gradually increases again during the week until the Weekend. The difference between the busiest day and the quietest day is around 56%. This does not happen to members. The difference between the days of week is minimum, nothing more than 13%.
* The average of ride length for Casual users is of 2771 seconds, which means 46 minutes, while for members is only 850 seconds, or 14 minutes.
* The last finding is that Sunday is the day with more long rides for both groups, while Tuesday is the day with less rides distances.


8. Act phase

Targeted Marketing Campaigns: The data shows that casual riders tend to ride longer distances and spend more time riding than members. This insight can be used to develop targeted marketing campaigns highlighting the flexibility and convenience of casual riding, emphasizing the ability to ride for longer durations without the commitment of an annual membership.

Pricing and Subscription Strategies: The data reveals that members perform more rides than casual users. This information can inform pricing and subscription strategies, such as offering discounted rates for frequent riders or introducing flexible membership options for those who ride occasionally. Adjusting pricing and subscription models to cater to the usage patterns of both casual riders and members can encourage more conversions.

Enhancing User Experience: The finding that both categories of users prefer docked bikes suggests that ensuring an adequate supply of docked bikes in popular areas can improve the user experience. By analyzing the data on popular ride times and locations, Cyclistic can optimize bike availability and distribution to meet user demand.

Customer Retention: Although the focus is on converting casual riders into members, it's essential to maintain a positive relationship with existing members. The data indicates that members ride consistently throughout the week, while casual riders show variations in their riding patterns. By providing targeted incentives and rewards for members, Cyclistic can enhance member loyalty and reduce churn.

Operational Efficiency: Analyzing the busiest and quietest days for both user categories can help optimize resource allocation and operational efficiency. By aligning staffing levels, bike maintenance schedules, and marketing efforts to match demand fluctuations, Cyclistic can improve overall operational performance.

Partnerships and Collaborations: By identifying the preferred ride days and average ride length for each user category, Cyclistic can explore partnerships with local businesses or event organizers to offer targeted promotions and incentives on specific days. This can encourage ridership and attract new customers, including casual riders who may be more inclined to use the service during certain events or weekends.

